src/plugins/simple-pricing/mutations/updateProductVariantPrices.js

```diff
@@ -35,24 +44,13 @@ export default async function updateProductVariantPrices(context, input) {
     legacyRoles: ["createProduct", "product/admin", "product/update"]
   });
 
-  const product = await Products.findOne({ _id: variantId, shopId });
-  if (!product) throw new ReactionError("not-found", "Product variant not found");
-
-  const updateDocument = { ...prices };
-
-  inputSchema.validate(updateDocument);
-
-  await Products.updateOne(
-    {
-      _id: variantId,
-      shopId
-    },
-    {
-      $set: updateDocument
-    }
+  const { value: updatedProduct } = await Products.findOneAndUpdate(
+    { _id: variantId, shopId, type: "variant" },
+    { $set: { ...prices }},
+    { returnOriginal: false }
   );
 
-  const updatedProduct = Products.findOne({ _id: variantId, shopId });
+  if (!updatedProduct) throw new ReactionError("error-occurred", "Unable to update variant prices");
 
   appEvents.emit("afterVariantUpdate", { productId: variantId, product: updatedProduct });
```

Comment:
@kieckhafer It looks like the correct way to emit this as is currently done in reaction admin is

```
appEvents.emit("afterVariantUpdate", { _id, field, value });
```

Where `_id` is the variant ID and it is called once per field that was changed.

I also noticed that in `src/core-services/catalog/startup.js`, it is looking for `productId`, which we never even emit, and using it as the variant ID (not product). BUT it actually just uses it to GET the product ID. So...

Can you?
- Change this to emit once per changed field (price and/or compareAtPrice)
- Emit with this signature: `{ _id: variantId, productId: updatedProduct.ancestors[0], variant: updatedProduct, field, value }`
- Change the event consumer in `src/core-services/catalog/startup.js` to this:

```js
appEvents.on("afterVariantUpdate", async ({ productId }) => {
    hashRelatedProduct(productId, collections).catch((error) => {
      Logger.error("Error in afterVariantUpdate", error);
    });
  });
```

Reply:
@aldeed I think this takes care of it: https://github.com/reactioncommerce/reaction/pull/5910/commits/92a296609daf0eeeaf8630c1e5764c4c5749bcbf

Not sure if looping over the object like that is the best way to do it though.